src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -517,10 +517,12 @@ protected override bool IsInvocableExpression(SyntaxNode node)
                 return true;
             }
 
+            // When a SimpleMemberAccess's ParentKind is either Invocation or ObjectionCreation, both the SimpleMemberAccess expression and the 
+            // parent expression act on the same Symbol. However, when the ParentKind is ElementAccess, the ElementAccess expression acts on an 
+            // indexer symbol that is different than member symbol referenced by the SimpleMemberAccess, so we should treat it as invocable.
             if (node.IsKind(SyntaxKind.SimpleMemberAccessExpression) &&
                 !node.IsParentKind(SyntaxKind.InvocationExpression) &&
-                !node.IsParentKind(SyntaxKind.ObjectCreationExpression) &&
-                !node.IsParentKind(SyntaxKind.ElementAccessExpression))
```

Comment:
💭 This change makes me uncomfortable, mostly because it's not clear to me what `IsInvocableExpression` is supposed to do. I'm going to mark this a Request Changes so we can get better information about this method added to either confirm this change or show that it's not what we want.